streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsMetricsImpl.java

```diff
@@ -0,0 +1,229 @@
+        return metrics.sensor(sensorName(operationName, entityName), recordLevel, parents);
+    }
+
+    @Override
+    public void removeSensor(String name) {
+        metrics.removeSensor(name);
+    }
+
+    private String groupNameFromScope(String scopeName) {
+        return "stream-" + scopeName + "-metrics";
+    }
+
+    private String sensorName(String operationName, String entityName) {
+        if (entityName == null) {
+            return prefix + "." + operationName;
+        } else {
+            return prefix + "." + entityName + "-" + operationName;
+        }
+    }
+
+    private Map<String, String> tagMap(String... tags) {
+        // extract the additional tags if there are any
+        Map<String, String> tagMap = new HashMap<>(this.tags);
+        if ((tags.length % 2) != 0)
+            throw new IllegalArgumentException("Tags needs to be specified in key-value pairs");
+
+        for (int i = 0; i < tags.length; i += 2)
+            tagMap.put(tags[i], tags[i + 1]);
+
+        return tagMap;
+    }
+
+
+
+    /**
+     * @throws IllegalArgumentException if tags is not constructed in key-value pairs
+     */
+    @Override
+    public Sensor addLatencySensor(String scopeName, String entityName, String operationName, Sensor.RecordLevel recordLevel, String... tags) {
+        Map<String, String> tagMap = tagMap(tags);
+
+        // first add the global operation metrics if not yet, with the global tags only
+        Sensor parent = metrics.sensor(sensorName(operationName, null), recordLevel);
+        addLatencyMetrics(scopeName, parent, "all", operationName, tagMap);
+
+        // add the operation metrics with additional tags
+        Sensor sensor = metrics.sensor(sensorName(operationName, entityName), recordLevel, parent);
+        addLatencyMetrics(scopeName, sensor, entityName, operationName, tagMap);
+
+        return sensor;
+    }
+
+    /**
+     * @throws IllegalArgumentException if tags is not constructed in key-value pairs
+     */
+    @Override
+    public Sensor addThroughputSensor(String scopeName, String entityName, String operationName, Sensor.RecordLevel recordLevel, String... tags) {
+        Map<String, String> tagMap = tagMap(tags);
+
+        // first add the global operation metrics if not yet, with the global tags only
+        Sensor parent = metrics.sensor(sensorName(operationName, null), recordLevel);
+        addThroughputMetrics(scopeName, parent, "all", operationName, tagMap);
+
+        // add the operation metrics with additional tags
+        Sensor sensor = metrics.sensor(sensorName(operationName, entityName), recordLevel, parent);
+        addThroughputMetrics(scopeName, sensor, entityName, operationName, tagMap);
+
+        return sensor;
+    }
+
+    @Override
+    public Sensor addCacheSensor(String scopeName, String entityName, String operationName, Sensor.RecordLevel recordLevel,  String... tags) {
```

Comment:
This is not introduced in this PR, but I'm wondering if there is any value we add this specific `addCacheSensor` instead of just using the "generic adding sensor" functions since we have already introduced them in this PR. Afterwards there is only one place using it for recording `cache hits` with `min, max, avg` which to me is quite generic usage.